src/main/java/seedu/address/model/task/Task.java

```
@@ -34,15 +44,33 @@ public Note getNote() {
         return note;
     }
 
+    public Status getStatus() {
+        return status;
+    }
+
+    /**
+     * Update the Status
+     */
+    public Task markDone() {
+        return new Task(title, note, new Status(Status.TaskStatus.DONE));
+    }
+
+    /**
+     * Update the Status
+     */
+    public Task markNotDone() {
+        return new Task(title, note, new Status(Status.TaskStatus.NOT_DONE));
+    }
+
     /**
-     * Returns true if both tasks have the same title and note.
+     * Returns true if both tasks have the same title, note and status.
      */
     public boolean isSameTask(Task otherTask) {
         return this.equals(otherTask);
     }
```

Should we allow users to add 2 tasks with the same title, same note, but different statuses? Then, the user cannot change the status as doing so will result in duplicate tasks.

Steps:
1. Add a task 
2. Mark that task (1st task) as done
3. Add another task with the same title and note as task in Step 1
4. Mark 2nd task as done

Expectation: Should not allow task in Step 3 to be added

Actual: Allows task at Step 3 to be added
App throws exception after Step 4 
(seedu.address.model.task.exceptions.DuplicateTaskException: Operation would result in duplicate tasks)

By the way, I think this exception is not handled by the MarkTaskCommand/UnmarkTaskCommand.

---

> Should we allow users to add 2 tasks with the same title, same note, but different statuses? 

@m1oojv What is the resolution to this issue? I would prefer if we don't support such behaviour. (i.e. Count two tasks as the same iff they have the same Title and Note)